### PR TITLE
Add support for DHCPv4

### PR DESCRIPTION
**Pull request by tzneal**
Started by Vasiliy Tolstov <v.tolstov@selfip.ru>

### CODE REVIEW

**Comment by vtolstov** [comment]
Thanks!

**Comment by tzneal** [comment]
Thanks, I didn't mean for those to be exported, fixed now.

**Comment by tzneal** [comment]
Is there something else needed here?

**Comment by lhausermann** [comment]
Sorry for the delay of answer. The only point I see is why do we need to allocate a buffer for decoding DHCP Options. Please explain or fix it using a slice. 
I'll merge right away.

File: layers/dhcpv4.go [hunk 1]
```diff
@@ -0,0 +1,563 @@
...
+			if err := o.encode(data[offset:]); err != nil {
+				return err
+			}
+			offset += 2 + len(o.Data)
+		}
+		optend := NewDHCPOption(DHCPOptEnd, nil)
+		if err := optend.encode(data[offset:]); err != nil {
+			return err
+		}
+	}
```
Comment (tzneal) [inline layers/dhcpv4.go]: Unless I'm missing something, I already fixed that here.  The options encode directly to the data buffer.

Comment (lhausermann) [inline layers/dhcpv4.go]: I was not talking about Encode() but DecodeFromBytes() which is still making a new buffer and write in it.

Comment (tzneal) [inline layers/dhcpv4.go]: Sorry, I missed that, fixed now.

**Pull request merged**